cpp/src/arrow/buffer.h

```diff
@@ -137,6 +139,25 @@ class ARROW_EXPORT Buffer {
   /// \return a new Buffer instance
   static std::shared_ptr<Buffer> FromString(std::string data);
 
+  /// \brief Construct an immutable buffer that takes ownership of the contents
+  /// of an std::vector (without copying it).
+  ///
+  /// \param[in] data a string to own
+  /// \return a new Buffer instance
+  template <typename T>
+  static std::shared_ptr<Buffer> FromVector(std::vector<T> vec) {
```

Comment:
I would call this `FromPrimitiveVector` because people frequently mistakenly assume converting an `std::vector<std::string>` to an Arrow array is an straightforward operations.

And would it be OK to declare `vec` as `&&` so accidental copies aren't to easy to create?

Reply:
I'll rename and add a static assert for triviality of T.

I would prefer not to forbid copies just to prevent them from happening accidentally. This method would not be used in a performance critical context anyway. We build buffers with BufferBuilder in perf-critical contexts; I anticipate this mostly being useful in tests where requiring a call to something like `kj::cp` would get annoying.

Reply:
Fair enough.

Reply:
I have to accept that C++ can't prevent accidental copies. One has to remain vigilant at all times :)